tests/model_registry/async_job/utils.py

```diff
@@ -0,0 +1,89 @@
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.job import Job
+from ocp_resources.pod import Pod
+
+from tests.model_registry.async_job.constants import (
+    ASYNC_JOB_ANNOTATIONS,
+    ASYNC_JOB_LABELS,
+    MODEL_SYNC_CONFIG,
+    VOLUME_MOUNTS,
+)
+
+
+def get_job_pod(admin_client: DynamicClient, job: Job) -> Pod:
+    """Get the Pod created by a Job"""
+    pods = list(
+        Pod.get(
+            dyn_client=admin_client,
+            namespace=job.namespace,
+            label_selector=f"job-name={job.name}",
+        )
+    )
+    assert len(pods) == 1, f"Expected 1 pod for job {job.name}, found {len(pods)}"
+    return pods[0]
+
+
+def validate_job_labels_and_annotations(job: Job) -> None:
+    """Validate Job has all required labels and annotations"""
+    job_labels = job.instance.metadata.labels or {}
```

Comment:
Save `job.instance.metadata`, that way you don't need to do two api calls for labels and annotations.